### PR TITLE
Various fixes for Tribler 7

### LINKED ISSUE #3298: SQL error when adding torrent to session
See the following crash report:

```
Traceback (most recent call last):
File "/usr/share/tribler/TriblerGUI/event_request_manager.py", line 99, in on_read_data
raise RuntimeError(json_dict["event"]["text"])
RuntimeError: [Failure instance: Traceback: <class 'apsw.SQLError'>: SQLError: table Torrent has no column named status
/usr/lib/python2.7/dist-packages/twisted/internet/base.py:1194:run
/usr/lib/python2.7/dist-packages/twisted/internet/base.py:1203:mainLoop
/usr/lib/python2.7/dist-packages/twisted/internet/base.py:825:runUntilCurrent
/usr/lib/python2.7/dist-packages/twisted/internet/task.py:239:__call__
--- <exception caught here> ---
/usr/lib/python2.7/dist-packages/twisted/internet/defer.py:150:maybeDeferred
/usr/share/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py:507:_task_process_alerts
/usr/share/tribler/Tribler/Core/Libtorrent/LibtorrentMgr.py:338:process_alert
/usr/share/tribler/Tribler/Core/Libtorrent/__init__.py:17:invoke_func
/usr/share/tribler/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py:537:process_alert
/usr/share/tribler/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py:621:on_metadata_received_alert
/usr/share/tribler/Tribler/dispersy/util.py:37:helper
/usr/share/tribler/Tribler/Core/RemoteTorrentHandler.py:169:save_torrent
/usr/share/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py:314:addExternalTorrent
/usr/share/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py:422:_addTorrentToDB
/usr/share/tribler/Tribler/Core/CacheDB/sqlitecachedb.py:332:insert
/usr/share/tribler/Tribler/Core/CacheDB/sqlitecachedb.py:316:execute_write
/usr/share/tribler/Tribler/dispersy/util.py:46:helper
/usr/share/tribler/Tribler/dispersy/util.py:225:blockingCallFromThread
/usr/share/tribler/Tribler/Core/CacheDB/sqlitecachedb.py:285:execute
]
```

### LINKED ISSUE #3314: Metainfo not a dict
See this error from the crash reporter:

```
Traceback (most recent call last):
File "TriblerGUI/event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "multiprocessing/process.py", line 114, in run

File "TriblerGUI/core_manager.py", line 65, in start_tribler_core

File "twisted/internet/base.py", line 1199, in run

File "twisted/internet/base.py", line 1208, in mainLoop

--- <exception caught here> ---
File "twisted/internet/base.py", line 828, in runUntilCurrent

File "Tribler/dispersy/util.py", line 150, in wrapper

File "Tribler/Core/TFTP/handler.py", line 192, in _process_callbacks

File "Tribler/Core/TFTP/handler.py", line 261, in <lambda>

File "Tribler/dispersy/util.py", line 37, in helper

File "Tribler/Core/RemoteTorrentHandler.py", line 601, in _on_download_successful

File "Tribler/Core/TorrentDef.py", line 105, in load_from_memory

File "Tribler/Core/TorrentDef.py", line 119, in _create

File "Tribler/Core/Utilities/utilities.py", line 29, in validTorrentFile

exceptions.ValueError: metainfo not dict
```

Simply catching the ValueError in the TFTP response callback and logging it should be sufficient I think.

### LINKED ISSUE #3320: Invalid torrents might not have the info field in the metainfo
See the following crash report:

```
Traceback (most recent call last):
File "/usr/share/tribler/TriblerGUI/event_request_manager.py", line 99, in on_read_data
raise RuntimeError(json_dict["event"]["text"])
RuntimeError: [Failure instance: Traceback: <type 'exceptions.KeyError'>: 'info'
/usr/lib/python2.7/dist-packages/twisted/web/resource.py:250:render
/usr/share/tribler/Tribler/Core/Modules/restapi/torrentinfo_endpoint.py:125:render_GET
/usr/lib/python2.7/dist-packages/twisted/internet/defer.py:321:addCallback
/usr/lib/python2.7/dist-packages/twisted/internet/defer.py:310:addCallbacks
--- <exception caught here> ---
/usr/lib/python2.7/dist-packages/twisted/internet/defer.py:653:_runCallbacks
/usr/share/tribler/Tribler/Core/Modules/restapi/torrentinfo_endpoint.py:63:on_got_metainfo
]
```

I'm unable to find the corrupt torrent on the internet, unfortunately. Should be easy to reproduce, however.

### LINKED ISSUE #3321: No tracker info available leading to error
See the following crash report:

```
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "lib\multiprocessing\process.py", line 114, in run

File "TriblerGUI\core_manager.py", line 65, in start_tribler_core

File "lib\site-packages\twisted\internet\base.py", line 1243, in run

File "lib\site-packages\twisted\internet\base.py", line 1252, in mainLoop

--- <exception caught here> ---
File "lib\site-packages\twisted\internet\base.py", line 878, in runUntilCurrent

File "Tribler\Core\TorrentChecker\torrent_checker.py", line 115, in _task_select_tracker

File "Tribler\Core\Modules\tracker_manager.py", line 81, in update_tracker_info

exceptions.TypeError: 'NoneType' object has no attribute '__getitem__'
```

### LINKED ISSUE #3322: Error when attempting to decode mugshot in settings
See the following crash report:

```
Traceback (most recent call last):
File "TriblerGUI\tribler_window.py", line 377, in received_settings
RuntimeError: 'utf8' codec can't decode byte 0x7b in position 0: could not dump:
dict[settings]->dict[general]->dict[mugshot]->tuple->str::'\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00\xff\xdb\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07\t\t\x08\n\x0c\x14\r\x0c\x0b\x0b\x0c\x19\x12\x13\x0f\x14\x1d\x1a\x1f\x1e\x1d\x1a\x1c\x1c $.\' ",#\x1c\x1c(7),01444\x1f\'9=82<.342\xff\xdb\x00C\x01\t\t\t\x0c\x0b\x0c\x18\r\r\x182!\x1c!2222222222222222
```

### LINKED ISSUE #3323: Uncatched int conversion in GUI settings
See the following crash report:

```
Traceback (most recent call last):
File "TriblerGUI\widgets\settingspage.py", line 176, in save_settings
ValueError: invalid literal for int() with base 10: '0,1'
```

### PR DESCRIPTION
**Pull request by devos50**
Fixes #3298, fixes #3314, fixes #3320, fixes #3321, fixes #3322, fixes #3323

### CODE REVIEW

**Comment by devos50** [comment]
@xoriole could you review this PR?

File: TriblerGUI/widgets/settingspage.py [hunk 1]
```diff
@@ -192,11 +192,17 @@ def save_settings(self):
...
-                                                               * 1024
+        try:
+            if self.window().upload_rate_limit_input.text():
+                settings_data['libtorrent']['max_upload_rate'] = int(self.window().upload_rate_limit_input.text()) \
+                                                                 * 1024
+            if self.window().download_rate_limit_input.text():
+                settings_data['libtorrent']['max_download_rate'] = int(self.window().download_rate_limit_input.text()) \
+                                                                   * 1024
+        except ValueError:
+            ConfirmationDialog.show_error(self.window(), "Invalid number for maximum upload/download",
```
Comment (qstokkink) [inline TriblerGUI/widgets/settingspage.py]: You can give the user a bit more information here, something like:

```python
def get_illegal_characters(input)
    return reduce(lambda x, y: x + ('' if y.isdigit() else y), input, "")

print "Found the following illegal characters", get_illegal_characters(input)
```

Comment (devos50) [reply TriblerGUI/widgets/settingspage.py]: I've clarified the error a bit in such a way that the user is informed that a whole number is required. I think this error message should just be a small amount of text.

**Comment by devos50** [comment]
I assume this PR is ready now? Then I will merge it 👍

**Pull request merged**